great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py

```diff
@@ -219,48 +231,59 @@ def get_expect_values_to_be_between_chart(
+            df=df,
+            metric_plot_util=metric_plot_util,
+            domain_plot_util=domain_plot_util,
+            batch_id_plot_util=batch_id_plot_util,
+            min_value_plot_util=min_value_plot_util,
+            max_value_plot_util=max_value_plot_util,
+            subtitle=subtitle,
+        )
 
-        title: Union[str, alt.TitleParams] = f"{metric_title} per {domain_title}"
-        if subtitle:
-            title = alt.TitleParams(title, subtitle=[subtitle])
+    @staticmethod
+    def _get_expect_values_to_be_between_chart(
+        df: pd.DataFrame,
+        metric_plot_util: MetricPlotUtil,
+        domain_plot_util: DomainPlotUtil,
+        batch_id_plot_util: BatchIdPlotUtil,
+        min_value_plot_util: PlotUtil,
+        max_value_plot_util: PlotUtil,
+        subtitle: Optional[str],
+    ) -> alt.Chart:
+        line_color: alt.HexColor = alt.HexColor(ColorPalettes.HEATMAP_6.value[4])
 
-        batch_id: str = "batch_id"
-        batch_id_title: str = batch_id.replace("_", " ").title().replace("Id", "ID")
-        batch_id_type: alt.StandardType = AltairDataTypes.NOMINAL.value
-        min_value: str = "min_value"
-        min_value_title: str = min_value.replace("_", " ").title()
-        min_value_type: alt.StandardType = AltairDataTypes.QUANTITATIVE.value
-        max_value: str = "max_value"
-        max_value_title: str = max_value.replace("_", " ").title()
-        max_value_type: alt.StandardType = AltairDataTypes.QUANTITATIVE.value
+        title: alt.TitleParams = DataAssistantResult._determine_plot_title(
+            metric_title=metric_plot_util.title,
+            domain_title=domain_plot_util.title,
+            subtitle=subtitle,
+        )
 
         tooltip: List[alt.Tooltip] = [
-            alt.Tooltip(field=batch_id, type=batch_id_type, title=batch_id_title),
-            alt.Tooltip(
-                field=metric_name, type=metric_type, title=metric_title, format=","
-            ),
-            alt.Tooltip(
-                field=min_value, type=min_value_type, title=min_value_title, format=","
-            ),
-            alt.Tooltip(
-                field=max_value, type=max_value_type, title=max_value_title, format=","
-            ),
+            batch_id_plot_util.generate_tooltip(),
+            metric_plot_util.generate_tooltip(format=","),
+            min_value_plot_util.generate_tooltip(format=","),
+            max_value_plot_util.generate_tooltip(format=","),
         ]
 
         lower_limit: alt.Chart = (
             alt.Chart(data=df)
             .mark_line(color=line_color)
             .encode(
-                x=alt.X(
-                    domain_name,
-                    type=domain_type,
-                    title=domain_title,
-                ),
-                y=alt.Y(min_value, type=metric_type, title=metric_title),
+                x=domain_plot_util.plot_on_axis(),
+                y=metric_plot_util.plot_on_axis(shorthand=min_value_plot_util.name),
```

Comment:
I believe what you are calling `shorthand` here is actually the required parameter `field`. The `shorthand` usually refers to a string with the field, and/or type, and/or aggregate (which you correctly described in the docstring for `plot_on_axis`). So if an actual shorthand is passed here I'm not sure what the override behavior is in the `alt.Y` object because you already set the type and title in the method using `self`.

Are we intending for the user to be able to override the Altair data type here? I think I understand why you use `metric_plot_util` and then override the name with `min_value_plot_util.name`, but why wouldn't we just use `y=min_value_plot_component.plot_on_axis()` here?

Reply:
the docs seem to say it's the shorthand? https://altair-viz.github.io/user_guide/generated/channels/altair.X.html

Reply:
also the reason why I passed in that value instead of running `min_value_plot_component.plot_on_axis()` is because we're mixing metric and min_value attrs.

This was the previous code:
```
y=alt.Y(min_value, type=metric_type, title=metric_title),
```
the `type` and `title` correspond to the `metric_plot_component` but `min_value` has to come from `min_value_plot_component`.